include/exadg/operators/inverse_mass_operator.h

```diff
@@ -46,6 +49,10 @@ struct InverseMassOperatorData
   unsigned int quad_index;
 
   InverseMassParameters parameters;
+
+  // Variable parameter.
+  bool                                                          coefficient_is_variable;
+  VariableCoefficients<dealii::VectorizedArray<Number>> const * variable_coefficients;
```

Comment:
This stores a scalar per quadrature point; in combination with the mass operator we keep two separate arrays that need to be kept in sync. I would have implemented the operation by using the data of the mass operator (through some access functionality), and instead apply the inverse operation here by a division rather than multiplication in line 264-265. Of course, a division incurs some cost, especially when running from caches, but I would introduce that option as soon as measurements indicate we need it. Or what application settings do you envision, and would it ever be problematic to fill both separately (as far as I see, that code is not there yet)?

Reply:
I tried to make it more similar to `ViscousOperator` now, but the difference is that the `ViscousOperator` holds `ViscousOperatorData` and a shared_ptr to the `ViscousKernel`, while
the mass operator owns the Masskernel and the MassOperatorData now.

I suggest we discuss the new commit (incoming) and then check if we have to refine the design later on, once all the parts are in place!

Reply:
yes thanks, makes sense!
the code here actually uses the coefficient^-1, so is tailored to assembling
(p, 1/coefficient p)
which is not general at all, but from the present PR this is not clear.
We will name the coefficients here `variable_coefficients` and add a comment saying that one gets the action of 
(p, coefficient p)^-1
Then we add getters to the `mass_kernel`/`mass_operator` and only need to update one single place!
Then we probably add some parameter `invert_coefficient` to signal if we want `coefficient` or `1/coefficient`.

I will look into this.